### PR TITLE
[JENKINS-51793] explicit enable

### PR DESCRIPTION
**Pull request by mwinter69**
Explicitly enable indexing
When a 1.x version of the plugin was installed but never configured and then one updates to 2.0
the configuration is initially not saveable.
Also when installing the first time one couldn't save without
configuring the plugin. With the enable option it is initially disabled
and one can save without problems.

### CODE REVIEW

File: src/main/java/jenkins/plugins/logstash/LogstashConfiguration.java [hunk 1]
```diff
@@ -45,9 +46,30 @@
...
+      {
+        enabled = true;
+      }
+    }
     activeIndexer = logstashIndexer;
   }
 
+  public boolean isEnabled()
+  {
+    return enabled;
```
Comment (jakub-bochenski) [inline src/main/java/jenkins/plugins/logstash/LogstashConfiguration.java]: potential NPE here, maybe use Optional<Boolean> instead?

File: src/main/java/jenkins/plugins/logstash/LogstashConsoleLogFilter.java [hunk 2]
```diff
@@ -26,6 +26,12 @@ public LogstashConsoleLogFilter(Run<?, ?> run)
   @Override
   public OutputStream decorateLogger(Run build, OutputStream logger) throws IOException, InterruptedException
   {
+    LogstashConfiguration configuration = LogstashConfiguration.getInstance();
+    if (!configuration.isEnabled())
+    {
+      return logger;
```
Comment (jakub-bochenski) [inline src/main/java/jenkins/plugins/logstash/LogstashConsoleLogFilter.java]: can we add a log entry (fine?) here, just to notify whats happening?

File: src/main/java/jenkins/plugins/logstash/LogstashNotifier.java [hunk 3]
```diff
@@ -88,6 +88,12 @@ public void perform(Run<?, ?> run, FilePath workspace, Launcher launcher,
   }
 
   private boolean perform(Run<?, ?> run, TaskListener listener) {
+    LogstashConfiguration configuration = LogstashConfiguration.getInstance();
+    if (!configuration.isEnabled())
+    {
```
Comment (jakub-bochenski) [inline src/main/java/jenkins/plugins/logstash/LogstashNotifier.java]: can we add a log entry (fine?) here, just to notify whats happening?

File: src/test/resources/disabled.xml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+<jenkins.plugins.logstash.LogstashConfiguration plugin="logstash@1.4.0-SNAPSHOT">
+  <logstashIndexer class="jenkins.plugins.logstash.configuration.RabbitMq">
+    <host>localhost</host>
+    <port>5672</port>
+    <queue>logstash</queue>
+    <username></username>
+    <password>{AQAAABAAAAAQ5wlElDfWGri9VuaMh0MCBZWs1fjL31zvnxrkszfW5pA=}</password>
+  </logstashIndexer>
+  <enabled>false</enabled>
```
Comment (jakub-bochenski) [inline src/test/resources/disabled.xml]: I think this tag would not be present in an actual migration from 1.4 (seeing that you are adding the field now)?

File: src/main/java/jenkins/plugins/logstash/LogstashConfiguration.java (hunk 1 above)
Comment (mwinter69) [reply src/main/java/jenkins/plugins/logstash/LogstashConfiguration.java]: This should be covered by the constructor. It is checking after `load()` if enabled is set.

File: src/test/resources/disabled.xml (hunk 4 above)
Comment (mwinter69) [reply src/test/resources/disabled.xml]: This file is not used for testing migration.

File: src/main/java/jenkins/plugins/logstash/LogstashConfiguration.java (hunk 1 above)
Comment (jakub-bochenski) [reply src/main/java/jenkins/plugins/logstash/LogstashConfiguration.java]: I see, but this is still very easy to break. Can we add an explicit guard here if you don't want to use optional (`enabled == null ? false : enabled`)?

BTW strange that FB doesn't flag this as potential NPE

File: src/test/resources/disabled.xml (hunk 4 above)
Comment (jakub-bochenski) [reply src/test/resources/disabled.xml]: OK then, `plugin="logstash@1.4.0-SNAPSHOT"` confused me

File: src/main/java/jenkins/plugins/logstash/LogstashConfiguration.java (hunk 1 above)
Comment (mwinter69) [reply src/main/java/jenkins/plugins/logstash/LogstashConfiguration.java]: Probably FB assumes due to Constructor that it can't be null.

**Pull request merged**